Updated BusyIndicator Integration topic (#855)
telerik/maui-docs: controls/busyindicator/integration.md

---

> title: Integration

Reviewer: I am not entirely convinced this one should be a page on its own rather than a how-to KB. What's the case for having a separate page?

Furthermore, the Integration title in the navigation might mislead users that this is going to be a more general document, rather than a very specific integration. If you decide to keep this as a separate article (rather than a KB), change the title to `CollectionView Integration`.

page_title: .NET MAUI BusyIndicator Documentation - CollectionView Integration

> description: Learn more about the option to integrate BusyIndacator with CollectionView control.

Reviewer: ```suggestion
description: Learn more about the option to integrate the BusyIndicator with the CollectionView control.
```

position: 2

> previous_url: /controls/busyindicator/integrate-with-listview

Reviewer: You need to add in the old URL from the deleted page as well.

```suggestion
previous_url: /controls/busyindicator/integrate-with-listview,/controls/busyindicator/busyindicator-integrate-with-listview
```

slug: busyindicator-integration
---

# .NET MAUI BusyIndicator Integration with CollectionView

> The Telerik BusyIndicator for .NET MAUI is useful when you want to display a notification to the end-users of the application while a long-running operation, such as loading data from a service, is currently in progress.

Reviewer: Only use `end-user` as an adjective or adverb. Here, you need to go with `end users`.

```suggestion
The Telerik BusyIndicator for .NET MAUI is useful when you want to display a notification to the application's end users during a long-running operation. For example, you might need to display a busy indicator while the application is loading data from a service.
```

The example below demonstrates a sample integration of the BusyIndicator with the CollectionView control. The CollectionView loads its data asynchronously (this is simulated for the purpose of the example) and while the load operation is taking place, the `IsBusy` state of the BusyIndicator is enabled.

**1.** Create a sample `Book` class used for the `ItemsSource` of the CollectionView:

<snippet id='busyindicator-withcollectionview-model' />

**2.** Add a `ViewModel` class, which provides the following:

  * A collection of `Book` objects that is used for binding the CollectionView.
  * A Boolean `IsLoading` property to control the Busy state of the BusyIndicator.
  * A `LoadData` command that starts the loading of the items.

<snippet id='busyindicator-withcollectionview-csharp' />

**3.** Add the CollectionView and the BusyIndicator controls to the view:

<snippet id='busyindicator-withcollectionview-xaml' />

**4.** Set the `ViewModel` class as `BindingContext` of the page:

<snippet id='busyindicator-withcollectionview-setvm' />

The gif below shows the result.

![Telerik UI for .NET MAUI BusyIndicator Integration](images/busyindicator-integration.gif)

## See Also

> - [BusyIndicator Animations]({% slug busyindicator-animations %})

Reviewer: Add links to the CollectionView overview page and to the BusyIndicator overview page.
